build/gen_stub.php

```diff
@@ -59,6 +59,23 @@ function processStubFile(string $stubFile, Context $context): ?FileInfo {
             echo "Saved $arginfoFile\n";
         }
 
+        if ($context->generateMethodsynopses && $fileInfo->generateMethodsynopses) {
+            $methodsynopsesDirectory = dirname($stubFile) . "/methodsynopses";
+
+            $methodsynopses = generateMethodsynopses($fileInfo);
+            if (!empty($methodsynopses) && ($context->forceRegeneration || $stubHash !== $oldStubHash)) {
+                if (!file_exists($methodsynopsesDirectory)) {
+                    mkdir($methodsynopsesDirectory);
+                }
+
+                foreach ($methodsynopses as $filename => $content) {
+                    if (file_put_contents("$methodsynopsesDirectory/$filename", $content)) {
```

Comment:
I don't think that we want to 'xinclude' these files; while that would be convenient regarding potential regeneration, it would make editing the docs harder (consider that e.g. parameter names need to be changed in other places on the page as well). Still, single files might be preferable for the import into the docs. I'll have to check that out.

Reply:
@cmb69 Did you come to a conclusion with this? I think I'll try to implement a regex-based automatic synopsis & parameter name replacer tool.

Reply:
The issue is how we can do the mapping between the generated methodsynopses and the respective manual pages. I have still no idead how.

Anyhow, did you consider some XML tooling instead of regex based tool?

Reply:
No, not yet :) Regex is just my default tool for these kind of problems. I don't know yet if it is really suitable for the job.